### ISSUE TITLE
Simple way convert row to JsonObject?

### ISSUE THREAD

**Issue by daihy8759** [eclipse-vertx/vertx-sql-client#353]
Dear Authors,

Is there has Simple way to convert JsonObject from row?

**Comment by vietj** [comment]
you mean a json object with row names as keys ?

**Comment by vietj** [comment]
not yet but there will be some mapping to JsonObject / DataObject in Vert.x 4 at least

**Comment by daihy8759** [comment]
yeah,The conversion method is a bit complicated now, some data types I have not done yet.
```
List<JsonObject> list = new ArrayList<>();
for (Row row : rows) {
    int size = row.size();
    JsonObject jsonObject = new JsonObject();
    for (int i = 0; i < size; i++) {
        String columnName = row.getColumnName(i);
        Object value = row.getValue(i);
        if(value != null){
            if(value instanceof LocalDateTime){
                jsonObject.put(columnName, ((LocalDateTime)value).atZone(
                        ZoneId.systemDefault()).toInstant());
            } else {
                jsonObject.put(columnName, row.getValue(i));
            }
        }
    }
    list.add(jsonObject);
}
return list;
```

**Comment by decalek** [comment]
In many cases, json serialization of the resultset/resultsets goes outside of the app server (e.g. to web client or service consumer) without any mangling.

IMHO, In these cases, it may be feasible JsonGenerator to be used for direct row serialization to the stream.

(Sorry if this is already aranged in vert.x, I not have learned the platform yet)

**Comment by rgmz** [comment]
> yeah,The conversion method is a bit complicated now, some data types I have not done yet.
> ...

This is presently what I am doing as well:

```kotlin
fun RowSet<Row>.toJson(): List<JsonObject> = asSequence().map { row ->
    val json = JsonObject()
    columnsNames().asSequence().forEach {
        json.put(it, toPgType(row.getValue(it)))
    }
    json
}.toList()

private fun toPgType(value: Any?): Any? = when (value) {
    is String -> value
    is LocalDate -> value.format(DateTimeFormatter.ISO_DATE)
    is LocalDateTime -> value.format(DateTimeFormatter.ISO_DATE_TIME)
    else -> value
}
```

**Comment by vietj** [comment]
in Vert.x 4 we will try to improve this so one can store a date directly in JSON Object

**Comment by tobiasmuehl** [comment]
> > yeah,The conversion method is a bit complicated now, some data types I have not done yet.
> > ...
> 
> This is presently what I am doing as well:
> 
> ```kotlin
> fun RowSet<Row>.toJson(): List<JsonObject> = asSequence().map { row ->
>     val json = JsonObject()
>     columnsNames().asSequence().forEach {
>         json.put(it, toPgType(row.getValue(it)))
>     }
>     json
> }.toList()
> 
> private fun toPgType(value: Any?): Any? = when (value) {
>     is String -> value
>     is LocalDate -> value.format(DateTimeFormatter.ISO_DATE)
>     is LocalDateTime -> value.format(DateTimeFormatter.ISO_DATE_TIME)
>     else -> value
> }
> ```

If you use postgres `TIMESTAMP WITH TIMEZONE` type, you also need this type mapping:

```kotlin
        is OffsetDateTime -> value.format(DateTimeFormatter.ISO_DATE_TIME)
```

**Comment by tsegismont** [comment]
Outdated, tracked by #754 and solved in 4.0.0
